saplings/circuits/src/components/circuitDetails/CircuitDetails.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2018-2020 Cargill Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../Content.scss';
+import '../MainHeader.scss';
+import './CircuitDetails.scss';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faQuestionCircle,
+  faArrowLeft
+} from '@fortawesome/free-solid-svg-icons';
+import PropTypes from 'prop-types';
+import { useParams, Link } from 'react-router-dom';
+import React from 'react';
+
+import { Circuit } from '../../data/circuits';
+import { useCircuitState } from '../../state/circuits';
+import { getNodeRegistry } from '../../api/splinter';
+
+const CircuitDetails = () => {
+  const { circuitId } = useParams();
+  const [circuit] = useCircuitState(circuitId);
+
+  if (!circuit) {
+    return <div />;
+  }
+  return (
+    <div>
+      <div className="main-header">
+        <div className="back-button">
+          <Link to="/circuits">
+            <FontAwesomeIcon icon={faArrowLeft} />
+            &nbsp;Circuits
+          </Link>
+        </div>
+        <div className="circuits-title">
+          <h4>
+            Circuit
+            {circuitId}
+          </h4>
+          <div className="managementType">
+            Gameroom
+            <span>
+              <FontAwesomeIcon icon={faQuestionCircle} />
+            </span>
+          </div>
+        </div>
+      </div>
+      <div className="main-content">
+        <div className="midContent">
+          <div className="circuit-stats">
+            <div className="stat total-circuits">
+              <span className="stat-count circuits-count">
+                {circuit.members.length}
+              </span>
+              Nodes
+            </div>
+            <div className="stat action-required">
+              <span className="stat-count action-required-count">
+                {circuit.roster.length}
+              </span>
+              Services
+            </div>
+          </div>
+        </div>
+
+        <NodesTable circuit={circuit} nodes={circuit.members} />
+      </div>
+    </div>
+  );
+};
+
+const NodesTable = ({ circuit, nodes }) => {
+  const [fullNodes, setNodes] = React.useState(null);
+
+  React.useEffect(() => {
+    const fetchNodes = async () => {
```

Comment:
You could use the `useNodeRegistryState` in `state/nodeRegistry` instead of repeating the request here. Is there a reason you did not use it?

Reply:
The effect, in this case, is based off the change of `nodes`, which is actually a collection of node ID values.  They operate differently. I could modify the  node registry state to be a reducer, but that seemed a bit overly complex.